File: snippets/fsharp/System/NullReferenceException/Overview/example1.fs
module Example

// <Snippet1>
open System

[<EntryPoint>]
let main args =
    let value = Int32.Parse args[0]
    // Set names to null, don't initialize it. 
    let mutable names = Unchecked.defaultof<ResizeArray<string>>
    if value > 0 then
        names <- ResizeArray()
    names.Add "Major Major Major"
    0
// Compilation does not display a warning as this is an extremely rare occurance in F#.
// Creating a value without initalizing either requires using 'null' (not possible
// on types defined in F# without [<AllowNullLiteral>]) or Unchecked.defaultof.
//
// The example displays output like the following output:
//    Unhandled Exception: System.NullReferenceException: Object reference
//    not set to an instance of an object.
//       at Example.main()
// </Snippet1>

File: snippets/csharp/System/NullReferenceException/Overview/example1.cs
// <Snippet1>
using System.Collections.Generic;

public class UseBeforeAssignExample
{
    public static void Main(string[] args)
    {
        int value = int.Parse(args[0]);
        List<string> names;
        if (value > 0)
            names = [];

        //names.Add("Major Major Major");
    }
}
// Compilation displays a warning like the following:
//    Example1.cs(10) : warning BC42104: Variable //names// is used before it
//    has been assigned a value. A null reference exception could result
//    at runtime.
//
//          names.Add("Major Major Major")
//          ~~~~~
// The example displays output like the following output:
//    Unhandled Exception: System.NullReferenceException: Object reference
//    not set to an instance of an object.
//       at Example.Main()
// </Snippet1>
